File: mobile/src/main/java/xyz/mechenbier/circuittester/UiUpdateReceiver.java
package xyz.mechenbier.circuittester;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/**
 * Created by mechenbier on 8/1/2017.
 */

class UiUpdateReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {

        if (intent.hasExtra(context.getString(R.string.charging_intent_charging_extra_name))){
            boolean isCharging = intent.getBooleanExtra(context.getString(R.string.charging_intent_charging_extra_name),false);
            MainActivity.getInstance().setChargingIconColor(isCharging);
        }
    }
}

File: mobile/src/main/java/xyz/mechenbier/circuittester/UiUpdateReceiver.kt
package xyz.mechenbier.circuittester

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

/**
 * Created by mechenbier on 8/1/2017.
 */
internal class UiUpdateReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {

        if (intent.hasExtra(INTENT_POWER_CONNECTION_RECEIVER_CHARGING_EXTRA_ISCHARGING)) {
            val isCharging = intent.getBooleanExtra(INTENT_POWER_CONNECTION_RECEIVER_CHARGING_EXTRA_ISCHARGING, false)
            MainActivity.mInstance.setUiFromChargingState(isCharging)
        }
    }
}